File: sdks/js/examples/test-collector.ts
import { Quota } from '../src/index';

/**
 * Exemplo prático do SDK Quota em TypeScript / JavaScript
 */
async function runExample() {
  console.log('=== Teste do SDK Quota (@quota/sdk) ===\n');

  // 1. Inicializa o SDK com as configurações globais
  Quota.init({
    apiKey: 'qta_live_test_123456', // Substiua pela sua API Key do Quota
    endpoint: 'http://localhost:3000/collector',
    project: 'projeto-suporte',
    agent: 'bot-financeiro',
    environment: 'development',
    debug: true
  });

  console.log('--- Teste 1: Envio Manual de Métricas (Quota.trackUsage) ---');
  await Quota.trackUsage({
    provider: 'openai',
    model: 'gpt-4o',
    promptTokens: 150,
    completionTokens: 45,
    totalTokens: 195,
    latencyMs: 320,
    statusCode: 200,
    metadata: {
      externalUserId: 'user_991823',
      requestGroup: 'fluxo-cobranca',
      tags: ['manual-tracking', 'teste']
    }
  });

  console.log('\n--- Teste 2: Interceptação Automática via Cabeçalhos Customizados ---');
  
  // Simulando como o SDK intercepta uma chamada de requisição de IA (ex: OpenAI)
  // Quando o usuário passar cabeçalhos x-quota-*, o escutador captura automaticamente!
  try {
    const fakeOpenAIResponse = await fetch('https://api.openai.com/v1/chat/completions', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer sk-fake-key-para-teste',
        'x-quota-agent': 'atendimento-humano',
        'x-quota-user-id': 'cliente_4412',
        'x-quota-tags': 'sac,ouvidoria'
      },
      body: JSON.stringify({
        model: 'gpt-4o',
        messages: [{ role: 'user', content: 'Qual o meu saldo?' }]
      })
    });

    console.log('Status da chamada simulada:', fakeOpenAIResponse.status);
  } catch (err) {
    // Como a chave da OpenAI era fake no teste, pode retornar erro HTTP normal da OpenAI
    console.log('Chamada concluída (interceptador executou em background).');
  }

  // Aguarda 1s para visualizar os logs assíncronos do microtask
  await new Promise((resolve) => setTimeout(resolve, 1000));
}

runExample();

File: sdks/python/examples/test_collector.py
import time
import sys
import os

# Adiciona o diretório pai ao sys.path para importar o pacote local 'quota'
sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))

from quota import Quota
import httpx
import requests

def run_example():
    print("=== Teste do SDK Quota em Python (quota-sdk) ===\n")

    # 1. Inicializa o SDK
    Quota.init(
        api_key="qta_live_test_py_123456",
        endpoint="http://localhost:3000/collector",
        project="saas-python",
        agent="bot-python-atendimento",
        environment="development",
        debug=True
    )

    print("\n--- Teste 1: Rastreamento Manual (Quota.track_usage) ---")
    Quota.track_usage({
        "provider": "openai",
        "model": "gpt-4o",
        "promptTokens": 180,
        "completionTokens": 60,
        "totalTokens": 240,
        "latencyMs": 410,
        "statusCode": 200,
        "metadata": {
            "externalUserId": "py_user_551",
            "tags": ["python", "manual"]
        }
    })

    print("\n--- Teste 2: Interceptação Automática de chamadas httpx com cabeçalhos ---")
    try:
        with httpx.Client() as client:
            res = client.post(
                "https://api.openai.com/v1/chat/completions",
                headers={
                    "Content-Type": "application/json",
                    "Authorization": "Bearer sk-fake-key",
                    "x-quota-agent": "agente-httpx-python",
                    "x-quota-user-id": "usr_httpx_99",
                    "x-quota-tags": "httpx,teste"
                },
                json={
                    "model": "gpt-4o",
                    "messages": [{"role": "user", "content": "Olá do Python!"}]
                }
            )
            print(f"Status do httpx simulado: {res.status_code}")
    except Exception as e:
        print("Chamada httpx concluída (interceptador capturou em background).")

    print("\n--- Teste 3: Interceptação Automática de chamadas requests com cabeçalhos ---")
    try:
        res = requests.post(
            "https://api.openai.com/v1/chat/completions",
            headers={
                "Content-Type": "application/json",
                "Authorization": "Bearer sk-fake-key",
                "x-quota-agent": "agente-requests-python",
                "x-quota-user-id": "usr_req_88"
            },
            json={
                "model": "gpt-4o",
                "messages": [{"role": "user", "content": "Teste requests Python"}]
            }
        )
        print(f"Status do requests simulado: {res.status_code}")
    except Exception as e:
        print("Chamada requests concluída (interceptador capturou em background).")

    # Aguarda 1 segundo para visualização dos logs da thread assíncrona
    time.sleep(1)

if __name__ == "__main__":
    run_example()
